tests/Builder/BuilderCodecTest.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace MongoDB\Tests\Builder;
+
+use Generator;
+use MongoDB\Builder\Aggregation;
+use MongoDB\Builder\BuilderEncoder;
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Query;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\TestCase;
+
+use function array_is_list;
+use function array_merge;
+use function array_walk;
+use function is_array;
+
+/**
+ * @todo This annotation is not enough as this PHP file needs to use named arguments, that can't compile on PHP 7.4
+ * @requires PHP 8.0
+ */
+class BuilderCodecTest extends TestCase
+{
+    /** @see https://www.mongodb.com/docs/manual/reference/operator/aggregation/match/#equality-match */
+    public function testPipeline(): void
+    {
+        $pipeline = new Pipeline(
+            // @todo array is accepted by the stage class, but we expect an object. The driver accepts both.
+            Stage::match((object) ['author' => 'dave']),
```

Comment:
Is the object-cast necessary here? The `match()` factory method and Match constructor only say they accept `mixed $query`, so I don't see anything that enforces the argument type.

Does that mean the implementation for Match is incomplete, or is that intentional? I'm curious what would happen if you passed a scalar value here. Would it just encode to an invalid stage like `{ "$match": 1 }`?

Reply:
It's a perverse effect to convert all associative arrays into object with `objectify` to improve readability of expected result. The value could be an associative array here.

Reply:
Are you saying that the `(object)` cast was just done to make the test assertion more readable (presumably in the event it fails)? I wasn't thinking of `objectify()` when I wrote the above comment and was more just generally curious about how `Stage::match()` reacts to an invalid value being passed in since it only uses `mixed`.

Reply:
Confirmed this was a temporary issue, and the type system will be improved down the line.